src/Aspire.Dashboard/Components/Pages/Resources.razor

```diff
@@ -10,25 +10,75 @@
-                      Immediate="true"
-                      @bind-Value="_filter"
-                      slot="end"
-                      @bind-Value:after="HandleSearchFilterChangedAsync" />
-    </FluentToolbar>
+@{
+    var showDetailsView = SelectedResource is not null;
+}
 
-    <FluentPopover AnchorId="typeFilterButton" @bind-Open="_isTypeFilterVisible" AutoFocus="true">
-        <Header>@Loc[nameof(Dashboard.Resources.Resources.ResourcesResourceTypesHeader)]</Header>
-        <Body>
+<AspirePageContentLayout IsSummaryDetailsViewOpen="@showDetailsView">
+    <PageTitleSection>
+        <h1 class="page-header">@Loc[nameof(Dashboard.Resources.Resources.ResourcesHeader)]</h1>
+    </PageTitleSection>
+
+    <ToolbarSection>
+        @if (ViewportInformation.IsDesktop)
+        {
+            <FluentButton id="typeFilterButton" slot="end"
+                          Appearance="@(AreAllTypesVisible is true ? Appearance.Stealth : Appearance.Accent)"
+                          IconEnd="@(new Icons.Regular.Size24.Filter())"
+                          @onclick="() => _isTypeFilterVisible = !_isTypeFilterVisible"
+                          Title="@(AreAllTypesVisible is true ? Loc[nameof(Dashboard.Resources.Resources.ResourcesTypeFilterAllVisible)] : Loc[nameof(Dashboard.Resources.Resources.ResourcesTypeFiltered)])"
+                          aria-label="@(AreAllTypesVisible is true ? Loc[nameof(Dashboard.Resources.Resources.ResourcesTypeFilterAllVisible)] : Loc[nameof(Dashboard.Resources.Resources.ResourcesTypeFiltered)])"/>
+        }
+        else
+        {
+            <div>
+                <h4>@(AreAllTypesVisible is true ? Loc[nameof(Dashboard.Resources.Resources.ResourcesTypeFilterAllVisible)] : Loc[nameof(Dashboard.Resources.Resources.ResourcesTypeFiltered)])</h4>
+
+                <FluentStack Orientation="Orientation.Vertical">
+                    <FluentCheckbox
+                        Label="@ControlsStringsLoc[nameof(ControlsStrings.All)]"
+                        ThreeState="true"
+                        ShowIndeterminate="false"
+                        ThreeStateOrderUncheckToIntermediate="true"
+                        @bind-CheckState="AreAllTypesVisible"/>
+                    @foreach (var (resourceType, _) in _allResourceTypes)
+                    {
+                        var isChecked = _visibleResourceTypes.ContainsKey(resourceType);
+                        <FluentCheckbox
+                            Label="@resourceType"
+                            @bind-Value:get="isChecked"
+                            @bind-Value:set="c => OnResourceTypeVisibilityChangedAsync(resourceType, c)"/>
+                    }
+                </FluentStack>
+            </div>
+        }
+
+
+            @if (ViewportInformation.IsDesktop)
+            {
+                <FluentSearch Placeholder="@ControlsStringsLoc[nameof(ControlsStrings.FilterPlaceholder)]"
```

Comment:
State is lost when resizing and switching between views.

Losing filtering state because you resized the browser is very unexpected.

![state-lost-on-resize](https://github.com/dotnet/aspire/assets/303201/55eb1821-7656-48de-bc0f-1c6a7033f9d2)

Reply:
We can fix state being lost easily. See the below comment for additional information

Reply:
This has been fixed